internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -727,3 +727,25 @@ func (r *Reconciler) reconcileReferencedTemplate(ctx context.Context, in reconci
 
 	return nil
 }
+
+// createErrorWithoutObjectName removes the name of the object from the error message. As each new Create call involves and
+// object with a unique generated name each error appears to be a different error. As the errors are being surfaced in a condition
+// on the Cluster, the name is removed here to prevent each creation error from triggering a new reconciliation.
+func createErrorWithoutObjectName(err error, obj client.Object) error {
+	out := ""
+	var statusError *apierrors.StatusError
+	if errors.As(err, &statusError) {
+		if statusError.Status().Details != nil {
+			for _, cause := range statusError.Status().Details.Causes {
+				out = fmt.Sprintf("%s %s: %s: %s", out, cause.Type, cause.Field, cause.Message)
+			}
+			out = fmt.Sprintf("%s.%s: %s.", statusError.Status().Details.Kind, statusError.Status().Details.Group, out)
+		}
+
+		// Replace the statusError message with the constructed message.
+		statusError.ErrStatus.Message = fmt.Sprintf("failed to create %s", out)
```

Comment:
```suggestion
		statusError.ErrStatus.Message = fmt.Sprintf("failed to create: %s", out)
```
nit, usually errors are concatenated with a colon as separator

Reply:
Should we only add `out` if it's not empty? Currently the error would have a trailing space in that case

Reply:
If out is empty at this point the error isn't going to be make any sense regardless I think - should we instead return the error on L750 if there's nothing in out?

Reply:
Yes!